### ISSUE TITLE
Feature Request: buildGoModule Support

### ISSUE THREAD

**Issue by diogox** [berberman/nvfetcher#51]
Hi :wave: 

`buildGoModule` requires a `vendorSha256` field to be built. I was wondering if this could be supported by nvfetcher like it does for Rust?

Thanks!

**Comment by diogox** [comment]
I'm not familiar with this project's code, but can't we run this `nix-prefetch` to get the vendorSha256?

```shell
nix-prefetch '{ sha256 }: golangci-lint.go-modules.overrideAttrs (_: { modSha256 = sha256; })'
```

Where `golangci-lint` is the name of the package? Of course this would not work for local packages as it stands :thinking:

**Comment by berberman** [comment]
> but can't we run this nix-prefetch to get the vendorSha256?

No, because nvfetcher runs before we have the derivation, whereas using `nix-prefetch` like that requires the derivation.

**Comment by NickCao** [comment]
vendorSha256 can be computed (or built) from `go.sum` alone, hust like with `Cargo.lock`, however the process is harder due the the obscure tree hashing scheme of go modules. A somehow working example can be found here: https://gist.github.com/NickCao/e2135fae47b31798c5d03cae1d242cb3

**Comment by NickCao** [comment]
To make the situation worse, `go.sum` generated by go pre 1.16 and go 1.17 differs slightly and is generally incompatible.

**Comment by NickCao** [comment]
With the introduction of impure derivations in recent versions of nix, the approach in https://gist.github.com/NickCao/e2135fae47b31798c5d03cae1d242cb3 is now feasible (without considering pouring tens of thousands of derivations into your nix store).

**Comment by bandithedoge** [comment]
In some cases you can pass `null` to `vendorSha256` to use vendored dependencies from your source repo. There is also the legacy [`buildGoPackage`](https://nixos.org/manual/nixpkgs/stable/#ssec-go-legacy), which requires `goDeps` generated with [go2nix](https://github.com/kamilchm/go2nix)
